lib/service/service.go

```diff
@@ -679,7 +679,15 @@ func NewTeleport(cfg *Config) (*TeleportProcess, error) {
 		} else {
 			switch cfg.JoinMethod {
 			case types.JoinMethodToken, types.JoinMethodUnspecified, types.JoinMethodIAM:
-				cfg.HostUUID = uuid.NewString()
+				// Checking error instead of the usual uuid.New() in case uuid generation
+				// fails due to not enough randomness.
```

Comment:
Please add this, so it's clear it does happen.

```suggestion
				// fails due to not enough randomness.
				// It's been known to happen happen when Teleport starts very early in
				// the node initialization cycle and /dev/urandom isn't ready yet
```